packages/playwright/src/matchers/toMatchText.ts

```diff
@@ -64,39 +64,30 @@ export async function toMatchText(
   const { matches: pass, received, log, timedOut } = await query(!!this.isNot, timeout);
   const stringSubstring = options.matchSubstring ? 'substring' : 'string';
   const receivedString = received || '';
-  const message = pass
-    ? () =>
-      typeof expected === 'string'
-        ? matcherHint(this, receiver, matcherName, 'locator', undefined, matcherOptions, timedOut ? timeout : undefined) +
-        `Expected ${stringSubstring}: not ${this.utils.printExpected(expected)}\n` +
-        `Received string: ${printReceivedStringContainExpectedSubstring(
-            receivedString,
-            receivedString.indexOf(expected),
-            expected.length,
-        )}` + callLogText(log)
-        : matcherHint(this, receiver, matcherName, 'locator', undefined, matcherOptions, timedOut ? timeout : undefined) +
-        `Expected pattern: not ${this.utils.printExpected(expected)}\n` +
-        `Received string: ${printReceivedStringContainExpectedResult(
-            receivedString,
-            typeof expected.exec === 'function'
-              ? expected.exec(receivedString)
-              : null,
-        )}` + callLogText(log)
-    : () => {
-      const labelExpected = `Expected ${typeof expected === 'string' ? stringSubstring : 'pattern'
-      }`;
-      const labelReceived = 'Received string';
+  const messagePrefix = matcherHint(this, receiver, matcherName, 'locator', undefined, matcherOptions, timedOut ? timeout : undefined);
+  const notFound = received === kNoElementsFoundError;
+  const message = () => {
+    if (pass) {
+      if (typeof expected === 'string') {
+        if (notFound)
+          return messagePrefix + `Expected ${stringSubstring}: not ${this.utils.printExpected(expected)}\nReceived: ` + received + callLogText(log);
+        const printedReceived = printReceivedStringContainExpectedSubstring(receivedString, receivedString.indexOf(expected), expected.length);
+        return messagePrefix + `Expected ${stringSubstring}: not ${this.utils.printExpected(expected)}\nReceived string: ${printedReceived}` + callLogText(log);
+      } else {
+        if (notFound)
```

Comment:
It seems like you can move `notFound` case outside of both branches.

Reply:
The expected label seems different.